### PR TITLE
Added Nested AG-Grid table(s) to search page to display search info

### PR DESCRIPTION
**Pull request by carsonwolber**
## Description

<what this pull request is doing>

-  this PR adds the AG-Grid code to display the results of a search, completing the MVP version of the search feature

<what will be done in later PRs and not included here>

- next PR will be the last for MVP, completing the 'average' and 'difference' columns on the profile page

## Milestones

- Users can view a list of other users matching their search and see a  breakdown of their profile and transaction stats

## Resources
<links to tutorials, code snippets, inspirations>
<if you took or translated specific code from the internet, please call it out here!>

I relied on [this](https://www.ag-grid.com/react-data-grid/master-detail-nesting/) doc to get the nested-table feature down 

## Test Plan
<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>

https://github.com/user-attachments/assets/dafb9555-3d67-4340-b93c-6ab3532728c2

### CODE REVIEW

File: frontend/HelperFuncs/utils.js [hunk 1]
```diff
@@ -249,3 +249,48 @@ export const getMatchingUsers = async(searchTerm) => {
     console.log(data)
     return data;
 }
+
+
+/*
+helper function to format the rows of the AG-Grid component on search pages
+*/
+export function getSearchRows(userData) {
+    console.log('userData', userData)
```
Comment (William-La) [inline frontend/HelperFuncs/utils.js]: extra log

Comment (carsonwolber) [reply frontend/HelperFuncs/utils.js]: this was deleted in the following pr on this branch

**Comment by William-La** [comment]
style nit: on the right side of the tables, the inner table ends at the same spot as the outer table whereas the left sides are different. This gives the inner table a "cut-off" look.

**Review by William-La — APPROVED** [review]
This search page looks great! Good job Carson.

**Comment by carsonwolber** [comment]
> style nit: on the right side of the tables, the inner table ends at the same spot as the outer table whereas the left sides are different. This gives the inner table a "cut-off" look.

fixed in daaec1d 
Screenshot 2024-07-15 at 2 18 53 PM
